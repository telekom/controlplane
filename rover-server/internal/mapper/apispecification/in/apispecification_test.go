// Copyright 2025 Deutsche Telekom IT GmbH
//
// SPDX-License-Identifier: Apache-2.0

package in

import (
	"github.com/gkampitakis/go-snaps/snaps"
	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"
)

var _ = Describe("ApiSpecification Mapper", func() {
	Context("MapRequest", func() {
		It("must map a ApiSpecificationUpdateRequest to an ApiSpecification correctly", func() {
			output, err := MapRequest(apiSpecification, resourceIdInfo)

			Expect(err).To(BeNil())

			Expect(output).ToNot(BeNil())
			snaps.MatchSnapshot(GinkgoT(), output)
		})

		It("must return an error if the input ApiSpecificationUpdateRequest is nil", func() {
			output, err := MapRequest(nil, resourceIdInfo)

			Expect(output).To(BeNil())

			Expect(err).ToNot(BeNil())
			Expect(err.Error()).To(ContainSubstring("input api specification is nil"))
		})

	})
})
