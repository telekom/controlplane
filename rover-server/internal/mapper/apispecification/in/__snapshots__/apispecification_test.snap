
[ApiSpecification Mapper MapRequest must map a ApiSpecificationUpdateRequest to an ApiSpecification correctly - 1]
&v1.ApiSpecification{
    TypeMeta:   v1.TypeMeta{Kind:"ApiSpecification", APIVersion:"rover.cp.ei.telekom.de/v1"},
    ObjectMeta: v1.ObjectMeta{
        Name:                       "rover-local-sub",
        GenerateName:               "",
        Namespace:                  "poc--poc--eni--hyperion",
        SelfLink:                   "",
        UID:                        "",
        ResourceVersion:            "",
        Generation:                 0,
        CreationTimestamp:          time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC),
        DeletionTimestamp:          (*v1.Time)(nil),
        DeletionGracePeriodSeconds: (*int64)(nil),
        Labels:                     {"cp.ei.telekom.de/environment":"poc"},
        Annotations:                {},
        OwnerReferences:            nil,
        Finalizers:                 nil,
        ManagedFields:              nil,
    },
    Spec:   v1.ApiSpecificationSpec{Specification:"info:\n    title: Rover API\n    version: 1.0.0\nopenapi: 3.0.0\nservers:\n    url: http://rover-api/eni/distr/v1\n"},
    Status: v1.ApiSpecificationStatus{},
}
---
